homeassistant/components/sensor/mqtt.py

```diff
@@ -104,6 +111,20 @@ def message_received(topic, payload, qos):
                 self._expiration_trigger = async_track_point_in_utc_time(
                     self.hass, self.value_is_expired, expiration_at)
 
+            if self._json_attributes:
+                self._attributes = {}
```

Comment:
Performance nitpick: Put this in the `except` clause to avoid setting the attribute twice.

Reply:
Sorry but not sure how to do this.  This feature will allow multiple attributes that could all be updated at different times, i.e. MQTT message 1 updates attribute 1 only, MQTT message 2 updates attribute 2 only, MQTT message 3 updates attribute 1 and 3 only

Reply:
Oh yes of course. Makes sense.